source/custom/nidcpower_service.custom.cpp

```diff
@@ -1,5 +1,56 @@
 #include <nidcpower/nidcpower_service.h>
 
+#include <stdexcept>
+
 namespace nidcpower_grpc {
 
+struct DriverErrorException : std::runtime_error {
+  DriverErrorException(int status) : std::runtime_error("") { status_ = status; }
+  int status_ = 0;
+};
+
+void CheckStatus(int status)
+{
+  if (status != 0) {
+    throw DriverErrorException(status);
+  }
+}
+
+::grpc::Status NiDCPowerService::MeasureMultiple(::grpc::ServerContext* context, const MeasureMultipleRequest* request, MeasureMultipleResponse* response)
+  {
+    if (context->IsCancelled()) {
+      return ::grpc::Status::CANCELLED;
+    }
+    try {
+      auto vi_grpc_session = request->vi();
+      ViSession vi = session_repository_->access_session(vi_grpc_session.id(), vi_grpc_session.name());
+      ViConstString channel_name = request->channel_name().c_str();
+
+      // Getting number of channels from ParseChannelCount API
```

Comment:
This and rest of the comments below just describe the code below. They don't add any information that can't be inferred by reading the code. We shouldn't add any comments that don't add value.

Reply:
Done removing comments.